crates/rover-client/src/introspection/schema.rs

```diff
@@ -0,0 +1,237 @@
+//! Schema encoding module used to work with Introspection result.
+//!
+//! More information on Schema Definition language(SDL) can be found in [this
+//! documentation](https://www.apollographql.com/docs/apollo-server/schema/schema/).
+//!
+use crate::query::graph::introspect;
+use sdl_encoder::{
+    Directive, EnumDef, EnumValue, Field, FieldValue, InputObjectDef, InputValue, InterfaceDef,
+    ObjectDef, ScalarDef, Schema as SDL, UnionDef,
+};
+use serde::Deserialize;
+use std::convert::TryFrom;
+
+pub type Introspection = introspect::introspection_query::ResponseData;
+pub type SchemaTypes = introspect::introspection_query::IntrospectionQuerySchemaTypes;
+pub type SchemaDirectives = introspect::introspection_query::IntrospectionQuerySchemaDirectives;
+pub type FullTypeFields = introspect::introspection_query::FullTypeFields;
+pub type FullTypeFieldArgs = introspect::introspection_query::FullTypeFieldsArgs;
+pub type __TypeKind = introspect::introspection_query::__TypeKind;
+
+// Represents GraphQL types we will not be encoding to SDL.
+const GRAPHQL_NAMED_TYPES: [&str; 13] = [
```

Comment:
Do you have a better name suggestion? I got `GRAPHQL_NAMED_TYPES` from [here](https://github.com/graphql/graphql-js/blob/main/src/utilities/printSchema.js#L56)

Reply:
`printFilteredSchema` allows you to pass in a `typeFilter`, which is used the filter named types. But `GRAPHQL_NAMED_TYPES` is specifically a list the named types you want to filter out. `graphql-js` splits these up into introspection types ([`isIntrospectionType`] (https://github.com/graphql/graphql-js/blob/540f33605ee047ab86e0d9666f4a60526d56ff0f/src/type/introspection.js#L531-L546)) and scalars the are included in the spec ([`isSpecifiedScalarType`](https://github.com/graphql/graphql-js/blob/540f33605ee047ab86e0d9666f4a60526d56ff0f/src/type/scalars.js#L272-L273)). It may make sense to follow that example and even use something like `isDefinedType` to check for types that are neither.

Reply:
Ah, right. I think filters can be introduced at a later stage. For now it's easier (and more efficient) to have these as const. (The filter function that will then be introduced should also have these as consts, as these values are explicitly evaluated at compile time)